packages/webapp/src/components/SensorReadingsLineChart/index.jsx

```diff
@@ -207,8 +227,16 @@ const PureSensorReadingsLineChart = ({
                   strokeWidth={2}
                   dataKey={attribute.value}
                   stroke={attribute.color}
-                  activeDot={{ r: 6 }}
                   isAnimationActive={false}
+                  activeDot={{ r: 5 }}
+                  dot={dataDensity(sensorReadings, attribute.value) < 0.9}
+                  type={
+                    dataDensity(sensorReadings, attribute.value) >= 0.9 ||
+                    (chartWidth < 400 && dataDensity(sensorReadings, attribute.value) > 0.5)
+                      ? 'natural'
+                      : 'linear'
+                  }
+                  connectNulls
```

Comment:
@kathyavini I am not sure I like the connected nulls myself as it doesn't necessarily reflect whats happening well. I forget if this was discussed.

<img width="610" alt="Screenshot 2023-05-24 at 9 52 42 AM" src="https://github.com/LiteFarmOrg/LiteFarm/assets/30075973/00adabf9-f038-491b-bd34-b3265282fbb9">
<img width="610" alt="Screenshot 2023-05-24 at 9 52 13 AM" src="https://github.com/LiteFarmOrg/LiteFarm/assets/30075973/e610c734-eb0b-46e6-b8ac-a646b431de04">

Reply:
Hmm! The connectNulls was used because I was asked to handle sparse data feeds (feeds with only 3-4 data points per day). Without connectNulls, no line is drawn at all and those feeds become a sequence of dots.

I'm leaning towards connectNulls being a property we just hard-code to certain feeds that we know can be sparse, but if you can think of the right conditional to check so we can conditionally apply it, we can do that instead! I can't think of one off the top of my head.

Reply:
Yeah true that makes sense. I wish we knew what kind of data to expect. I think with a dot density function that assessed the frequency/distribution of nulls could also be used to make the correct tickmarks. But that is too much for this PR. I am good with it!